src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.netcoreapp.cs

```diff
@@ -0,0 +1,332 @@
+                Action cleanup = () => Directory.CreateDirectory(dirName);
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, action, cleanup, expectedPaths);
+
+                action = () => Directory.Delete(dirNameSecond);
+                cleanup = () => Directory.CreateDirectory(dirNameSecond);
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, action, cleanup, expectedPaths);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_File_Create()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            using (var watcher = new FileSystemWatcher(testDirectory.Path))
+            {
+                string fileName = Path.Combine(testDirectory.Path, "file");
+                string secondFileName = Path.Combine(testDirectory.Path, "Secondfile");
+                string[] expectedPaths = new string[] { fileName, secondFileName };
+
+                watcher.Filters.Add(Path.GetFileName(fileName));
+                watcher.Filters.Add(Path.GetFileName(secondFileName));
+
+                Action action = () => File.Create(fileName).Dispose();
+                Action cleanup = () => File.Delete(fileName);
+
+                ExpectEvent(watcher, WatcherChangeTypes.Created, action, cleanup, expectedPaths);
+
+                action = () => File.Create(secondFileName).Dispose();
+                cleanup = () => File.Delete(secondFileName);
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Created, action, cleanup, expectedPaths);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_File_Delete()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            using (var watcher = new FileSystemWatcher(testDirectory.Path))
+            {
+                string fileName = Path.Combine(testDirectory.Path, "file");
+                string secondFileName = Path.Combine(testDirectory.Path, "Secondfile");
+                string[] expectedPaths = new string[] { fileName, secondFileName };
+
+                watcher.Filters.Add(Path.GetFileName(fileName));
+                watcher.Filters.Add(Path.GetFileName(secondFileName));
+
+                Action action = () => File.Delete(fileName);
+                Action cleanup = () => File.Create(fileName).Dispose();
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, action, cleanup, expectedPaths);
+
+                action = () => File.Delete(secondFileName);
+                cleanup = () => File.Create(secondFileName).Dispose();
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, action, cleanup, expectedPaths);
+            }
+        }
+
+        private void EqualItems(Collection<string> firstCollection, Collection<string> secondCollection)
```

Comment:
Why this instead of Assert.Equal?

Reply:
because watcher.Filters is Derived Class  from Collection<T> which leads to failure of the test

Reply:
Which overload was it picking up?

Reply:
```   
 System.IO.Tests.FileSystemWatcherFilterListTests.ClearFilters [FAIL]
        Assert.Equal() Failure
        Expected: Collection<String> []
        Actual:   NormalizedFilterCollection []
```
Similar is the case with ToString(). should we override ToString() ? is there any way to make the getType return collection<string> ? i tried using new keyword but its still returning ```NormalizedFilterCollection```

Reply:
Cast to `IEnumerable<string>` and it should work. This is the assert method you want: `public static void Equal<T>(IEnumerable<T> expected, IEnumerable<T> actual);` Don't write your own.